ConnectorRevit/ConnectorRevit/UI/ConnectorBindingsRevit.Events.cs

```diff
@@ -24,17 +24,32 @@ public partial class ConnectorBindingsRevit
 
   public override async void WriteStreamsToFile(List<StreamState> streams)
   {
-    await APIContext
-      .Run(app =>
-      {
-        using (Transaction t = new(CurrentDoc.Document, "Speckle Write State"))
+    try
+    {
+      await APIContext
+        .Run(app =>
         {
-          t.Start();
-          StreamStateManager.WriteStreamStateList(CurrentDoc.Document, streams);
-          t.Commit();
-        }
-      })
-      .ConfigureAwait(false);
+          using (Transaction t = new(CurrentDoc.Document, "Speckle Write State"))
+          {
+            t.Start();
+            StreamStateManager.WriteStreamStateList(CurrentDoc.Document, streams);
+            t.Commit();
+          }
+        })
+        .ConfigureAwait(false);
+    }
+    // note : I don't think this exception catch is necessary, but this is an async VOID instead
+    // of an async Task. Revit WILL CRASH if this method tries to throw an error, so keeping this
+    // catch out of an abundance of caution
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      SpeckleLog.Logger.Fatal(
+        ex,
```

Comment:
Perfect, this looks good to me. We can start checking SEQ for this and determine if something else is required later on.